src/Sulu/Bundle/ContentBundle/Preview/PreviewRenderer.php

```diff
@@ -89,6 +89,8 @@ public function render(PageBridge $content, $partial = false)
         // get controller and invoke action
         $request = new Request($query, $request, [], $cookies);
         $request->attributes->set('_controller', $content->getController());
+        $request->attributes->set('webspace', $content->getWebspaceKey());
+        $request->attributes->set('locale', $content->getLanguageCode());
```

Comment:
currently it will be done by the AdminRequestProcessor which also will be removed in the next PR i wont change here to much and as i said this will be removed soon.